### DISCUSSION TITLE
Language

### DISCUSSION THREAD

**Discussion by ramatomo** [echicken/lilygo-t5-clock#1] [Q&A]
Aside from texts, is there a way to display weather/time data in other languages?

**Comment by echicken** [comment]
For weather, it would be a change to how the OpenWeatherMap request URL is constructed on line 8 of [SimpleWeather.cpp](lib/SimpleWeather/SimpleWeather.cpp):

```cpp
_url = "/data/3.0/onecall?lat=" + String(lat) + "&lon=" + String(longi) + "&appid=" + Key +"&units=metric&exclude=minutely,hourly,daily,alerts";
```

We'd need to append for example `&lang=fr`, where `fr` could be any of the language codes listed [here](https://openweathermap.org/api/one-call-3#multi). Easy for anyone to hack that into their own fork of the project, but ideally we'd add a variable to the user-defined `src/config.h` as with the API key, latitude, longitude, etc.

As for the rest of the display, I didn't have localization in mind when I wrote this, so a bit of an overhaul would be needed.

**Reply by ramatomo** [reply]
`_url = "/data/3.0/onecall?lat=" + String(lat) + "&lon=" + String(longi) + "&lang=it" + "&appid=" + Key +"&units=metric&exclude=minutely,hourly,daily,alerts";`

I tried this but it remains in English`

**Reply by ramatomo** [reply]
> For weather, it would be a change to how the OpenWeatherMap request URL is constructed on line 8 of [SimpleWeather.cpp](lib/SimpleWeather/SimpleWeather.cpp):

checking better I got confused, it's not the weather that needs to be translated, but the date and the day of the week!
